tool/tsh/app.go

```diff
@@ -59,12 +70,23 @@ func onAppLogin(cf *CLIConf) error {
 			SessionID:   ws.GetName(),
 			PublicAddr:  app.GetPublicAddr(),
 			ClusterName: tc.SiteName,
+			AWSRoleARN:  arn,
 		},
 		AccessRequests: profile.ActiveRequests.AccessRequests,
 	})
 	if err != nil {
 		return trace.Wrap(err)
 	}
+
+	// Save information about the current AWS CLI access app. The CurrentAWSCLIApp is used during tsh aws call
+	// to load proper client app credentials.
+	tc.CurrentAWSCLIApp = app.GetName()
```

Comment:
Btw I just thought - this seems to imply that you can be logged into a single AWS app at the time but why this restriction?

Could it behave the same way as `tsh db connect` i.e. if you're logged into a single database, it would connect you to it by default, otherwise you need to provide it explicitly?

```bash
# logged into single aws app
tsh aws ec2 describe-instances

# logged into multiple aws apps
tsh aws --name=app2 ec2 describe-instances
```

We can leave this as a future improvement though.